adapters/repos/db/vector/dynamic/index.go

```diff
@@ -134,9 +133,6 @@ type dynamic struct {
 }
 
 func New(cfg Config, uc ent.UserConfig, store *lsmkv.Store) (*dynamic, error) {
-	if !entcfg.Enabled(os.Getenv("ASYNC_INDEXING")) {
-		return nil, errors.New("the dynamic index can only be created under async indexing environment")
-	}
```

Comment:
I believe that this is a required check and that dynamic indices don't work without async indexing. Can we check with AR whether this is still the case?